lib/Sema/CSGen.cpp

```diff
@@ -930,6 +930,16 @@ namespace {
       = { nullptr, nullptr };
     unsigned currentEditorPlaceholderVariable = 0;
 
+    /// Returns false and emits the specified diagnostic if the member reference
+    /// base is a nil literal. Returns true otherwise.
+    bool isValidBaseOfMemberRef(Expr *base, Diag<> diagnostic) {
+      if (auto nilLiteral = dyn_cast<NilLiteralExpr>(base)) {
```

Comment:
It's odd that we have this special case for subscripts but not other member references. Maybe we should just remove this and possibly handle the diagnostic elsewhere?

Reply:
I mentioned that in my previous comment - we are going to extend that to all references soon.

Reply:
Ah, I didn't see that, my bad.

Reply:
No problem :)